### PR TITLE
Create PropertiesClass for external properties (http.proxyHost, http.proxyPort, http.nonProxyHosts)

### PR DESCRIPTION
**Pull request by tvallin**
Sumarize supported java properties and create @PropertiesClass for it #4550

### CODE REVIEW

File: tests/e2e-core-common/src/test/java/org/glassfish/jersey/tests/e2e/common/TestExternalProperties.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        Response response = target("resource").request().get();
+
+        Assert.assertEquals(200, response.getStatus());
+        Assert.assertEquals("OK", response.readEntity(String.class));
+        Assert.assertFalse(proxyHit);
+    }
+
+    private void startFakeProxy() {
+        Server server = new Server(9997);
```
Comment (jansupol) [inline tests/e2e-core-common/src/test/java/org/glassfish/jersey/tests/e2e/common/TestExternalProperties.java:85]: new Server(PROXY_PORT)

File: tests/e2e-core-common/pom.xml [hunk 2]
```diff
@@ -43,6 +43,11 @@
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (jansupol) [inline tests/e2e-core-common/pom.xml:46]: Please create new proxy submodule in tests\integration\externalproperties\
and put the test there. Hopefully, more external properties would appear there

**Review by jansupol — CHANGES_REQUESTED** [review]

File: tests/integration/externalproperties/src/test/java/org/glassfish/jersey/tests/externalproperties/HttpProxyTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        try {
+            server.start();
+        } catch (Exception e) {
+
+        }
+    }
+
+    @Test
+    public void testProxy() {
+        System.setProperty(ExternalProperties.HTTP_PROXY_HOST, PROXY_HOST);
```
Comment (senivam) [inline tests/integration/externalproperties/src/test/java/org/glassfish/jersey/tests/externalproperties/HttpProxyTest.java:70]: why not to put those 2 properties to `@Before` method as well?
```
System.setProperty(ExternalProperties.HTTP_PROXY_HOST, PROXY_HOST);
System.setProperty(ExternalProperties.HTTP_PROXY_PORT, PROXY_PORT);
```

Comment (tvallin) [reply tests/integration/externalproperties/src/test/java/org/glassfish/jersey/tests/externalproperties/HttpProxyTest.java]: I think it makes sense !

**Review by jansupol — APPROVED** [review]

**Review by senivam — APPROVED** [review]

**Pull request merged**